pkg/machine/e2e/init_test.go

```diff
@@ -635,6 +637,94 @@ var _ = Describe("podman machine init", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(proc2Session.outputToString()).To(ContainSubstring("/proc/sys/fs/binfmt_misc/qemu-x86_64"))
 	})
+
+	It("init machine with invalid --provider for platform", func() {
+		var providerOverride string
+		switch testProvider.VMType() {
+		case define.QemuVirt, define.AppleHvVirt, define.LibKrun:
+			providerOverride = "wsl"
+		case define.WSLVirt, define.HyperVVirt:
+			providerOverride = "applehv"
+		default:
+			providerOverride = "unknown"
+		}
+		Expect(providerOverride).ToNot(Equal("unknown"))
+
+		i := initMachine{}
+		machineName := randomString()
+		session, err := mb.setName(machineName).setCmd(i.withImage(os.DevNull).withProvider(providerOverride)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session.errorToString()).To(ContainSubstring(fmt.Sprintf("Error: invalid provider %q override", providerOverride)))
+
+	})
+
+	It("darwin machine init with --provider", func() {
+		skipIfVmtype(define.QemuVirt, "Test is only for Darwin")
+		skipIfVmtype(define.WSLVirt, "Test is only for Darwin")
+		skipIfVmtype(define.HyperVVirt, "Test is only for Darwin")
+
+		i := initMachine{}
+		machine1 := randomString()
+		applehvSession, err := mb.setName(machine1).setCmd(i.withImage(os.DevNull).withProvider("applehv")).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(applehvSession.ExitCode()).To(Equal(0))
+
+		machine2 := randomString()
+		libkrunSession, err := mb.setName(machine2).setCmd(i.withImage(os.DevNull).withProvider("libkrun")).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(libkrunSession.ExitCode()).To(Equal(0))
+
+		list := new(listMachine)
+		list = list.withFormat("json")
+		listSession, err := mb.setCmd(list).run()
+		Expect(err).NotTo(HaveOccurred())
+		var listResponse []*entities.ListReporter
+		err = jsoniter.Unmarshal(listSession.Bytes(), &listResponse)
+		Expect(err).NotTo(HaveOccurred())
+		for _, l := range listResponse {
+			if l.Name == machine1 {
+				Expect(l.VMType).To(Equal("applehv"))
+				continue
+			}
+			if l.Name == machine2 {
+				Expect(l.VMType).To(Equal("libkrun"))
+			}
+		}
+	})
+
+	It("windows machine init with --provider", func() {
```

Comment:
This is a duplicate of the previous test, it would be much nicer if you could write it in a way where it doesn't copy/paste everything.